service/controller/v11/templates/ignition/cloud_provider_conf.go

```diff
@@ -9,6 +9,9 @@ location: {{ .Location }}
 aadClientId: {{ .AADClientID }}
 aadClientSecret: {{ .AADClientSecret }}
 {{- end}}
+cloudProviderRateLimit: true
+cloudProviderRateLimitQPS: 3
```

Comment:
Can this be fractional number? According to documentation here: https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-request-limits that config is reasonable if given k8s cluster is the only one using this specific service principal token, but in our case I believe we share it among all tenant clusters. Therefore the correct value would be rather something like `0.15` (=3/20).

Reply:
Yes it can be - I don't quite follow the logic for the `0.15` though. Can you expand?

Reply:
Here are some docs about the types and which options exist:
https://github.com/kubernetes/cloud-provider-azure/blob/master/docs/cloud-provider-config.md#cluster-config

Reply:
> I don't quite follow the logic for the 0.15 though. Can you expand?

My logic there is that number `3` (QPS) probably comes from this:
> Resource Manager allows up to 12,000 read requests per hour

Which calculates 12000/3600 == 3.33333. Now that's the absolute max for one service principle if I understand the azure documentation correctly. When we have say 20 tenant clusters on an installations and they all are sharing the same principles, it leaves single k8s with quota of 3.33333/20 which nicely rounded (with some room for error) lands on `0.15`.

Reply:
But this cloud provider thing is running on each node so it would have to be `3.3333 / Number of nodes` in that case.
And then I would have no idea what to put here :D

Reply:
Oh! Right. Then it must be rather like wtf.. `0.00375` (coming from 3 QPS / 20 clusters / 40 nodes per cluster) ?

Reply:
...but that allows then single kubelet to issue a query once every 266s (bit over 4 minutes..) - not sure if that works very well either...